### PR TITLE
Updated to V3

### PR DESCRIPTION
**Pull request by BrettRToomey**

### CODE REVIEW

**Comment by rasmusebbesen** [comment]
We need Bugsnag on all projects as well

File: README.md [hunk 1]
```diff
@@ -13,48 +13,64 @@ A Vapor template for convenient and fast scaffolding 🏎
...
+For MySQL you can either use a connection string (using the `DATABASE_URL` key) or you can supply hostname, username, password and database individually. These will be set automatically on Vapor Cloud, but they will need to be set when running locally.
 
-### Environment variables
+#### Redis
+
+Similar to MySQL, you can either use a connection string (using the `REDIS_URL` key) or you can supply hostname and database individually. These will be set automatically on Vapor Cloud, but they will need to be set when running locally.
+
+#### Mailgun
+
+In order to send out emails using Mailgun, the Mailgun environment variables needs to be set.
```
Comment (siemensikkema) [inline README.md]:
```suggestion
In order to send out emails using Mailgun, the Mailgun environment variables need to be set.
```

File: README.md [hunk 2]
```diff
@@ -13,48 +13,64 @@ A Vapor template for convenient and fast scaffolding 🏎
...
-    - `$DATABASE_HOSTNAME`: Hostname for MySQL (this will fallback to `127.0.0.1`).
-    - `$DATABASE_USER`: User for MySQL (this will fallback to `root`).
-    - `$DATABASE_PASSWORD`: Password for MySQL (this will fallback to 'root').
-    - `$DATABASE_DB`: Database name for project in MySQL (this will fallback to `my-project`).
-- `redis.json`
-    - `$REDIS_HOSTNAME`: Hostname for Redis (this will fallback to `127.0.0.1`).
-    - `$REDIS_DATABASE`: Redis database.
-    
-Environment variables related to MySQL and Redis will be set automatically if you use e.g. Vapor Cloud. All other variables needs to be set manually, e.g. through the CLI tool. For more information, have a look at the official [Vapor Cloud docs](docs.vapor.cloud).
+For generating tokens, a couple of signers needs to be set for JWT Keychain. These can be generated by using `openssl rand -base64 32`. Remember to generate different values for each environment.
```
Comment (siemensikkema) [inline README.md]:
```suggestion
For generating tokens, a couple of signers need to be set for JWT Keychain. These can be generated by using `openssl rand -base64 32`. Remember to generate different values for each environment.
```

File: Sources/App/Models/Entities/AdminPanelUser/Extensions/AdminPanelUser+NodesSSOAuthenticatable.swift [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import Vapor
+
+extension AdminPanelUser: NodesSSOAuthenticatable {
+    public static func authenticated(
+        _ user: AuthenticatedUser,
+        req: Request
+    ) -> Future<Response> {
+        return AdminPanelUser
+            .query(on: req)
+            .filter(\AdminPanelUser.email == user.email)
```
Comment (siemensikkema) [inline Sources/App/Models/Entities/AdminPanelUser/Extensions/AdminPanelUser+NodesSSOAuthenticatable.swift]:
```suggestion
            .filter(\.email == user.email)
```

File: Sources/App/Models/Entities/AdminPanelUser/Extensions/AdminPanelUser+NodesSSOAuthenticatable.swift [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                    return try AdminPanelUser(
+                        email: user.email,
+                        name: user.name,
+                        avatarURL: user.imageURL,
+                        role: AdminPanelUser.Role.superAdmin,
+                        password: try AdminPanelUser.hashPassword(randomPassword)
+                        ).save(on: req)
+                }
+
+                return Future.transform(to: existingUser, on: req)
```
Comment (siemensikkema) [inline Sources/App/Models/Entities/AdminPanelUser/Extensions/AdminPanelUser+NodesSSOAuthenticatable.swift]:
```suggestion
                return req.future(existingUser)
```

File: Sources/App/Setup/boot.swift [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import Sugar
+import Vapor
+
+/// Called after your application has initialized.
+public func boot(_ app: Application) throws {
+    // MARK: Leaf tags
+    // Add your tags like this:
+    // let tags = try app.make(MutableLeafTagConfig.self)
```
Comment (siemensikkema) [inline Sources/App/Setup/boot.swift]: Just to be sure, we're still recommending the MutableLeafTagConfig? Not the `use...Tags` pattern?

Comment (steffendsommer) [reply Sources/App/Setup/boot.swift]: Just pushed the update to use the `use...Tags` pattern (after all packages got updated) 🎉

**Review by steffendsommer — APPROVED** [review]

**Pull request merged**